packages/providers/mgt-msal2-provider/src/Msal2Provider.ts

```diff
@@ -0,0 +1,463 @@
+        typeof config.multiAccountDisabled !== 'undefined' ? config.multiAccountDisabled : true; //Set this to true for now. Once multi account is enabled, default will be false
+      try {
+        this._publicClientApplication.handleRedirectPromise().then((tokenResponse: AuthenticationResult | null) => {
+          this.handleResponse(tokenResponse);
+          if (tokenResponse === null) {
+            this.trySilentSignIn();
+          }
+        });
+      } catch (e) {
+        throw e;
+      }
+    } else {
+      throw new Error('clientId must be provided');
+    }
+
+    this.graph = createFromProvider(this);
+  }
+
+  /**
+   * Attempts to sign in user silently
+   *
+   * @memberof Msal2Provider
+   */
+  public async trySilentSignIn() {
+    let silentRequest: any = {
+      scopes: this.scopes,
+      domainHint: this._domainHint
+    };
+    if (this._sid || this._loginHint) {
+      silentRequest.sid = this._sid;
+      silentRequest.loginHint = this._loginHint;
+    } else {
+      this._account = this.getAccount();
+      if (this._account) {
+        silentRequest.sid = this._account.idTokenClaims.sid || null;
+        silentRequest.loginHint = this._account.idTokenClaims.preferred_username;
+      }
+    }
+    if (silentRequest.sid || silentRequest.loginHint) {
+      try {
+        this.setState(ProviderState.Loading);
+        const response = await this._publicClientApplication.ssoSilent(silentRequest);
+        if (response) {
+          this.handleResponse(response);
+        }
+      } catch (e) {
+        this.setState(ProviderState.SignedOut);
+      }
+    }
+  }
+
+  /**
+   * Log in the user
+   *
+   * @return {*}  {Promise<void>}
+   * @memberof Msal2Provider
+   */
+  public async login(): Promise<void> {
+    const loginRequest: PopupRequest = {
+      scopes: this.scopes,
+      loginHint: this._loginHint,
+      prompt: 'select_account',
+      domainHint: this._domainHint
+    };
+    if (this._loginType == LoginType.Popup) {
+      const response = await this._publicClientApplication.loginPopup(loginRequest);
+      this.handleResponse(response);
+    } else {
+      const loginRedirectRequest: RedirectRequest = { ...loginRequest };
+      this._publicClientApplication.loginRedirect(loginRedirectRequest);
+    }
+  }
+
+  /**
+   * Get all signed in accounts
+   *
+   * @return {*}
+   * @memberof Msal2Provider
+   */
+  public getAllAccounts() {
+    let usernames = [];
+    this._publicClientApplication.getAllAccounts().forEach((account: AccountInfo) => {
+      usernames.push({ username: account.username, id: account.homeAccountId });
+    });
+    return usernames;
+  }
+
+  /**
+   * Switching between accounts
+   *
+   * @param {*} user
+   * @memberof Msal2Provider
+   */
+  public switchAccount(user: any) {
+    this._publicClientApplication.setActiveAccount(this._publicClientApplication.getAccountByHomeId(user.id));
+    this.setStoredAccount();
+    this.fireActiveAccountChanged();
+  }
```

Comment:
the `fireActiveAccountChanged` can be called in the IProvider in the `switchAccount/setActiveAccount()` method. You can then override that method here and call `super.setActiveAccount()`. This way, you can make the fire method private

ex (showing with updated name and type):

```suggestion
  public setActiveAccount(account: IProviderAccount) {
    this._publicClientApplication.setActiveAccount(this._publicClientApplication.getAccountByHomeId(user.id));
    this.setStoredAccount();
    super.setActiveAccount(account);
  }
```